### PR TITLE
lotusrepo lens

### LINKED ISSUE #88: lotusrep lens fails to start due to missing dependencies
@lanzafame encountered this error when setting `VISOR_LENS=lotusrep`

Wrapped for clarity

```
{"level":"fatal","ts":"2020-10-12T20:49:45.770+1000","logger":"visor","caller":"sentinel-visor/main.go:116","msg":"setup storage
 and api: get node api: starting node: missing dependencies for function \"reflect\".makeFuncStub
 (/usr/lib/go/src/reflect/asm_amd64.s:12): missing types: lp2p.RawHost; host.Host; lp2p.BaseIpfsRouting; metrics.Reporter;
 *dtypes.ScoreKeeper; *stmgr.StateManager; *wallet.Wallet; *store.ChainStore; *stmgr.StateManager; *wallet.Wallet;
 *store.ChainStore; *wallet.Wallet; ffiwrapper.Verifier; *stmgr.StateManager; *store.ChainStore; beacon.Schedule;
 *stmgr.StateManager; *wallet.Wallet; *stmgr.StateManager; *wallet.Wallet; *stmgr.StateManager; *store.ChainStore; 
*messagepool.MessagePool; *messagesigner.MessageSigner; *dtypes.MpoolLocker; *stmgr.StateManager; *wallet.Wallet;
 *stmgr.StateManager; *wallet.Wallet; *store.ChainStore; *paychmgr.Manager; storagemarket.StorageClient; 
discovery.PeerResolver; retrievalmarket.RetrievalClient; *store.ChainStore; dtypes.ClientDataTransfer; host.Host; 
*stmgr.StateManager; *wallet.Wallet; *stmgr.StateManager; *store.ChainStore; *messagepool.MessagePool; 
*messagesigner.MessageSigner; *dtypes.MpoolLocker; *stmgr.StateManager; *store.ChainStore; *messagepool.MessagePool; 
*market.FundMgr; *stmgr.StateManager; *wallet.Wallet; *stmgr.StateManager; *store.ChainStore; *messagepool.MessagePool; 
*messagesigner.MessageSigner; *dtypes.MpoolLocker; *stmgr.StateManager; *wallet.Wallet; *stmgr.StateManager; 
*wallet.Wallet; *store.ChainStore; *paychmgr.Manager; *wallet.Wallet; ffiwrapper.Verifier; *stmgr.StateManager; 
*store.ChainStore; beacon.Schedule; *stmgr.StateManager; *wallet.Wallet; *wallet.Wallet; ffiwrapper.Verifier; 
*stmgr.StateManager; *store.ChainStore; beacon.Schedule; *stmgr.StateManager; *wallet.Wallet; *stmgr.StateManager; 
*store.ChainStore; *messagepool.MessagePool; *messagesigner.MessageSigner; *dtypes.MpoolLocker; *stmgr.StateManager; 
*wallet.Wallet; *slashfilter.SlashFilter; *chain.Syncer; *pubsub.PubSub; dtypes.NetworkName; beacon.Schedule"}
```

### PR DESCRIPTION
**Pull request by willscott**
This change switches the repo lens to directly create a chainstore, rather than spinning up a full internal lotus node. This (per [@ribasushi](https://github.com/ribasushi/ltsh/blob/5b0211033020570217b0ae37b50ee304566ac218/cmd/lotus-shed/deallifecycles.go#L41-L171)) should be lighter weight.

Need to audit usage of the lens to see if other modules than the chain one are used and need to also be filled in.

fix #88

### CODE REVIEW

File: lens/lotusrepo/repo.go [hunk 1]
```diff
@@ -3,23 +3,34 @@ package lotusrepo
...
 	"github.com/filecoin-project/lotus/lib/cachebs"
-	"github.com/filecoin-project/lotus/node"
+	"github.com/filecoin-project/lotus/lib/ulimit"
+	marketevents "github.com/filecoin-project/lotus/markets/loggers"
 	"github.com/filecoin-project/lotus/node/impl"
 	"github.com/filecoin-project/lotus/node/repo"
+	"github.com/filecoin-project/specs-actors/actors/runtime/proof"
 	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/ipfs/go-cid"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
```
Comment (ribasushi) [inline lens/lotusrepo/repo.go:28]: Wrong blockstore:
https://godoc.org/github.com/filecoin-project/lotus/lib/blockstore

Comment (willscott) [reply lens/lotusrepo/repo.go:28]: the definition an alias type though, right?

Comment (ribasushi) [reply lens/lotusrepo/repo.go:29]: I didn't dig hat you pulled: you might have just grabbed aliases and it is fine. But for future-self...

**Review by placer14 — APPROVED** [review]
LGTM

**Pull request merged**